src/udev/udev-event.c

```diff
@@ -549,7 +550,9 @@ static int on_spawn_io(sd_event_source *s, int fd, uint32_t revents, void *userd
 
         l = read(fd, p, size - 1);
         if (l < 0) {
-                if (errno != EAGAIN)
+                if (errno == EAGAIN)
+                        goto reenable;
+                else
```

Comment:
doesn't matter, but I'd drop the else, it's redundant given the first branch used "goto" anyway...

Reply:
Done